crates/emath/src/ts_transform.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{Pos2, Rect, Vec2};
+
+/// Linearly transforms positions via a translation, then a scaling.
+///
+/// [`TSTransform`] first translates points, then scales them with the scaling origin
+/// at `0, 0` (the top left corner)
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "bytemuck", derive(bytemuck::Pod, bytemuck::Zeroable))]
+pub struct TSTransform {
+    /// Scaling applied first, scaled around (0, 0).
+    pub scaling: f32,
+
+    /// Translation amount, applied after translation.
+    pub translation: Vec2,
+}
+
+impl Eq for TSTransform {}
+
+impl Default for TSTransform {
+    #[inline]
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl TSTransform {
+    pub const IDENTITY: Self = Self {
+        translation: Vec2::ZERO,
+        scaling: 1.0,
+    };
+
+    /// The translation is applied first, then scaling around 0, 0.
+    #[inline]
+    pub fn new(translation: Vec2, scaling: f32) -> Self {
```

Comment:
Wrong docstring again.

The point of applying scale first, is that the transform can also be seen as a pose.

So a scale of 3 means "make it 3 times as big" and a translation of [2,3] mean "put the center of the object at position [2, 3] in the other space".